### DISCUSSION TITLE
An OVMS version of software

### DISCUSSION THREAD

**Discussion by RichardPar** [meatpiHQ/wican-fw#56] [Ideas]
Any chances of getting OVMS version? - A GSM+GPS can be easily added to the current WiCAN-ODB

https://www.openvehicles.com/

Richard

**Comment by fowi4hjte** [comment]
This was not possible back then because the chipset used in the wican was unsupported by the outdated ISP-IDF 3.3 that was /is used by OVMS. Since Aug 2024 this changed somewhat and i also hope to see a OVMS firmware release for the wican or even better the wican pro hardware.

**Reply by meatpiHQ** [reply]
@fowi4hjte Why do you want to run OVMS on WiCAN, why not use their hardware?

Supporting OVMS would be possible it will take a lot of work, and I'm not sure how the OVMS team will feel about it, given my hardware is lower cost.

**Reply by fowi4hjte** [reply]
> @fowi4hjte Why do you want to run OVMS on WiCAN, why not use their hardware?

The WiCAN hardware have many benefits for my usecase. For example i plan to use WiCAN Pro with OVMS firmware. When i have to do some diagnostics to the car, i can just write different firmware to the WiCAN Pro and use its build in dedicated chip and use

    ISO 15765-4 CAN J1962
    SAE J1939 CAN J1962
    ISO 11898 raw CAN J1962
    Medium Speed CAN MS-CAN J1962
    GM high speed J1962
    Single Wire CAN GMLAN J1962
    SAE J1850-PWM
    SAE J1850-VPW
    ISO 9141-2
    ISO 14230-4 slow/fast

This is impossible with OVMS hardware.
The latest OVMS hardware can not use normal USB because its chipset is ESP32 and not something more modern. I am not a fan of the ESP32-S3 and would liked to see a RISC-V chip instead but ESP32-S3 is still better then the ESP32.

> Supporting OVMS would be possible it will take a lot of work, and I'm not sure how the OVMS team will feel about it, given my hardware is lower cost.

Its not about the price. Its about the benefits WiCAN have overall. Alone the ability to switch between two different projects with one hardware is a extreme benefit WiCAN have.
You do not know how the OVMS team would feel about it? Send a PR to OVMS with adding support for WiCAN Pro. The users can use this PR and the OVMS team would probably answer in this PR and tell how they feel about it.

**Reply by meatpiHQ** [reply]
@fowi4hjte Supporting OVMS is not that simple, it is doable but will require a lot of work. The main thing here is that the PRO speaks to the OBD chip using ELM327 protocol unlike the OVMS using ESP32 built in CAN and and external SPI channel (I think). The PRO has the built-in CAN connected, so it would be somewhat easy to support 1 channel.

To be honest I prefer to focus and Home assistant for now, because most of the project backers are interested in that.
